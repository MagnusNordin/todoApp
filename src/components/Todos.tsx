import { useAppSelector } from "../app/hooks"
import {
  selectPagimation,
  selectTodos,
  selectSelectedUser,
} from "../api/todoSlice"
import styled from "styled-components"

import { TodoItem } from "./TodoItem"

const TodoWrapper = styled.div`
  width: 100%;
`

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`
const HeaderItem = styled.div`
  margin: 0 50px;
`
export default function Todos() {
  const todos = useAppSelector(selectTodos)
  const pagination = useAppSelector(selectPagimation)
  const selectedUser = useAppSelector(selectSelectedUser)

  if (!todos) {
    return <>Laddar...</>
  }

  const userFilteredTodos =
    selectedUser >= 0
      ? todos.filter((todo) => todo.userId === selectedUser)
      : todos
  const filteredTodos = userFilteredTodos.slice(
    pagination.current * pagination.size,
    pagination.current * pagination.size + pagination.size,
  )
  return (
    <>
      <Header>
        <HeaderItem>Att göra</HeaderItem>
        <HeaderItem>Användare</HeaderItem>
      </Header>
      <TodoWrapper>
        {filteredTodos.map((element) => {
          return <TodoItem key={element.id} item={element} />
        })}
      </TodoWrapper>
    </>
  )
}
