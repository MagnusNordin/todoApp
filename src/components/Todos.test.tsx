import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../app/store"
import Todos from "./Todos"

test("todos", async () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Todos />
    </Provider>,
  )

  expect(asFragment).toMatchSnapshot()
})
