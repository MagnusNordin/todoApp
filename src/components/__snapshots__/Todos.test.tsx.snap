// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`todos 1`] = `[Function]`;
